examples/talk-llama/talk-llama.cpp

```diff
@@ -546,9 +549,16 @@ int main(int argc, char ** argv) {
 
                 // optionally give audio feedback that the current text is being processed
                 if (!params.heard_ok.empty()) {
-                    int ret = system((params.speak + " " + std::to_string(voice_id) + " '" + params.heard_ok + "'").c_str());
-                    if (ret != 0) {
-                        fprintf(stderr, "%s: failed to speak\n", __func__);
+                    std::ofstream speak_file(params.speak_file.c_str());
+                    if (speak_file.fail()) {
+                        fprintf(stderr, "%s: failed to open speak_file\n", __func__);
+                    } else {
+                        speak_file.write(params.heard_ok.c_str(), params.heard_ok.size());
+                        speak_file.close();
+                        int ret = system((params.speak + " " + std::to_string(voice_id) + " " + params.speak_file).c_str());
+                        if (ret != 0) {
+                            fprintf(stderr, "%s: failed to speak\n", __func__);
+                        }
```

Comment:
Let's factor out this as a helper function `speak_with_file()` in `common` and reuse it